src/Compilers/CSharp/Portable/Compiler/MethodCompiler.cs

```diff
@@ -971,7 +971,7 @@ public override object VisitField(FieldSymbol symbol, TypeCompilationState argum
                 }
 
                 // no need to emit the default ctor, we are not emitting those
-                if (methodSymbol.IsDefaultValueTypeConstructor(requireZeroInit: true))
+                if (methodSymbol.IsDefaultValueTypeConstructor(requireZeroInit: false))
```

Comment:
>requireZeroInit: false [](http://example.com/codeflow?start=63&length=22)

The logic here looks confusing. Why is it the right thing to ignore a non-zeroinit parameter-less constructor? Presumably the constructor does something useful beyond just zero initializing the instance.

Reply:
> The logic here looks confusing.

I agree, it is confusing. We're ignoring implicitly-declared parameterless constructors only though, not explicitly-declared constructors.

Reply:
> We're ignoring implicitly-declared parameterless constructors only though, not explicitly-declared constructors.

In that case the parameter must be true.

Reply:
It looks like either the parameter is no longer meaningful, or the logic here is incorrect.